radsat-sk/operation/subsystems/RDosimeter.c

```diff
@@ -1,111 +1,235 @@
 /**
  * @file RDosimeter.c
- * @date February 11, 2020
- * @author Tyrel Kostyk
+ * @date December 28, 2021
+ * @author Tyrel Kostyk (tck290) and Isaac Poirier (iap992)
  */
 
-#ifdef DONT_COMPILE_THE_FOLLOWING_CODE
+#include <RDosimeter.h>
+#include <RI2c.h>
+#include <string.h>
 
-// TODO: update this code to use real code instead of pseudocode
 
-
-//==============================================================================
-//                                   GLOBALS
-//==============================================================================
+/***************************************************************************************************
+                                            DEFINITIONS
+***************************************************************************************************/
 
 /**
- *  The slave addresses of the two Melanin-Dosimeter boards.
- *	TODO: confirm once actual addresses are determined
+ * Reference voltage for the Melanin-Dosimeter boards. Sensors return relative
+ * 8-bit values. E.g. 127 out of 255 would be 50%, representing 1.65V
  */
-uint8_t dosimeterBoardSlaveAddr[2] = {
+#define DOSIMETER_REFERENCE_VOLTAGE_MV	((float) 3300)
+
+/** Temperature scales slope  (m in y = mx + b) */
+#define TEMPERATURE_SCALE_SLOPE		((float) (1 / -13.6))
+/** Temperature scales offset (b in y = mx + b) */
+#define TEMPERATURE_SCALE_OFFSET	((float) 192.48)
+
+/** I2C Slave Address for Dosimeter Board One */
+#define DOSIMETER_1_I2C_SLAVE_ADDR	(0x4A)
+/** I2C Slave Address for Dosimeter Board Two */
+#define DOSIMETER_2_I2C_SLAVE_ADDR	(0x48)
+/** I2C Response for Dosimeter Board readings are 12 bits (2 bytes) long */
+#define DOSIMETER_I2C_DELAY			(0)
+
+/** I2C Response for Dosimeter Board readings are 12 bits (2 bytes) long */
+#define DOSIMETER_COMMAND_LENGTH			(1)
+/** I2C Response for Dosimeter Board readings are 12 bits (2 bytes) long */
+#define DOSIMETER_RESPONSE_LENGTH			(2)
+/** High byte of 12-bit readings must have top 4 bits masked away */
+#define DOSIMETER_RESPONSE_HIGH_BYTE_MASK	(0x0F)

```

Comment:
ARM processors are little-endian, so should this bitmask be flipped? I may be wrong as I haven't done a ton of bitwise ops recently.

Reply:
Nope; endianness only has to do with how memory is stored in the CPUs internal memory. From a programmer's perspective, masking and bitwise operations are essentially just arithmetic, which the CPU still understands appropriately.